src/lib.js

```diff
@@ -111,7 +119,7 @@ var program,
 
         if (validMethods.indexOf(curlObj.request.toUpperCase()) === -1) {
         // the method is still not valid
-          throw new Error('The method ' + curlObj.request + ' is not supported');
+          throw new UserError('The method ' + curlObj.request + ' is not supported');
```

Comment:
@dhwaneetbhatt Can we create a separate file containing the content of all such User Errors and export the content and use here?

This will help in managing the content better and readable for contributors with lesser context.